ros_ethercat_model/include/ros_ethercat_model/robot_state.hpp

```diff
@@ -88,12 +88,15 @@ class RobotState : public hardware_interface::HardwareInterface
            xit;
            xit = xit->NextSiblingElement("transmission"))
       {
-        std::string type = xit->Attribute("type");
-
-        Transmission *t = transmission_loader_.createUnmanagedInstance(type);
-        if (!t || !t->initXml(xit, this))
-          throw std::runtime_error(std::string("Failed to initialize transmission type: ") + type);
-        transmissions_.push_back(t);
+	if (xit->Attribute("type"))
```

Comment:
How about accepting either a type attribute or a type sub-element?
Sub-element is apparently the way it is defined now (http://wiki.ros.org/urdf/XML/Transmission).
Maybe we should migrate our hand definition to this format.
If this is not relevant for the current work (which I suspect) just ignore my comment.